src/librustdoc/config.rs

```diff
@@ -504,12 +504,18 @@ impl Options {
             return Err(1);
         }
 
-        let output = matches.opt_str("o").map(|s| PathBuf::from(&s)).unwrap_or_else(|| {
-            matches
-                .opt_str("output")
-                .map(|s| PathBuf::from(&s))
-                .unwrap_or_else(|| PathBuf::from("doc"))
-        });
+        let out_dir = matches.opt_str("o").map(|s| PathBuf::from(&s));
```

Comment:
nit: using `"o"` here seems confusing, I would rather use `out-dir` explicitly so it's more clear.
```suggestion
        let out_dir = matches.opt_str("out-dir").map(|s| PathBuf::from(&s));
```